src/radarChart.ts

```diff
@@ -677,29 +718,125 @@ module powerbi.extensibility.visual {
+            let shift: number = 0;
 
-                    return textMeasurementService.getTailoredTextOrDefault(properties, label.maxWidth);
-                },
-                labelLayout: {
-                    x: (label: RadarChartLabel) => label.x,
-                    y: (label: RadarChartLabel) => label.y,
-                },
-                filter: (label: RadarChartLabel) => (label != null),
-                style: {
-                    "font-size": PixelConverter.fromPoint(labelSettings.fontSize),
-                    "text-anchor": (label: RadarChartLabel) => label.textAnchor,
-                },
+            if (direction) {
+                shift = Math.abs(currentTextY12 - otherTextY21);
+            }
+            else {
+                shift = Math.abs(otherTexty22 - currentTextY1);
+            }
+
+            return shift * (direction ? 1 : -1);
+        }
+
+        private shiftIntersectText(current: RadarChartLabel, others: RadarChartLabel[], shiftDown: boolean): void {
+            let labelSettings: RadarChartLabelSettings = this.radarChartData.settings.labels;
+
+            let properties: TextProperties = {
+                fontFamily: RadarChart.AxesLabelsFontFamily,
+                fontSize: PixelConverter.fromPoint(labelSettings.fontSize),
+                text: this.radarChartData.labels.formatter.format(current.text)
             };
+
+            let currentTextHeight: number = textMeasurementService.estimateSvgTextHeight(properties);
+
+            for (let i: number = 0; i < others.length; i++) {
+                let label: RadarChartLabel = others[i];
+
+                properties.text = label.text;
+                let otherTextHeight: number = textMeasurementService.estimateSvgTextHeight(properties);
+
+                let curTextUpperPoint: number = current.y - currentTextHeight;
+                let labelTextUpperPoint: number = label.y - otherTextHeight;
+
+                if (this.isIntersect(current.y, curTextUpperPoint, label.y, labelTextUpperPoint)) {
+                    let shift: number = this.shiftText(current.y, curTextUpperPoint, label.y, labelTextUpperPoint, shiftDown);
+                    current.y += shift;
+                    if (!shiftDown && current.y < 0 || shiftDown && current.y + currentTextHeight / 2 > 0) {
+                        current.hide = true;
+                    }
+                }
+            }
+        }
+
+        private shiftCollidedLabels(labelPoints: RadarChartLabel[]): void {
+            // from 0 to 90 shift up by Y
+            let maxRadius: number = 0;
+            labelPoints.forEach(point => {
+                if (Math.abs(point.x) > maxRadius) {
+                    maxRadius = Math.abs(point.x);
+                }
+            });
+
+            let shiftDirrectionIsDown: boolean = this.radarChartData.settings.axisBeginning === 1;
+
+            for (let i: number = 0; i < labelPoints.length; i++) {
```

Comment:
consider refactoring the 2 for() loops below. There's some code repetition which can be refactored out for better readability.